torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -612,19 +627,52 @@ Value* emitBuiltinCall(
     Symbol name,
     at::ArrayRef<NamedValue> args,
     at::ArrayRef<NamedValue> kwargs,
-    const c10::optional<NamedValue>& self) {
+    const c10::optional<NamedValue>& self,
+    const c10::optional<int64_t> version) {
   const auto& variants = getAllOperatorsFor(name);
   const auto& builtin_functions = getAllBuiltinFunctionsFor(name);
 
+  // first let's set the graph's version
+  if (version.has_value()) {
+    if (graph.get_op_version().has_value()) {
+      TORCH_INTERNAL_ASSERT(version.value() == graph.get_op_version().value());
+    } else {
+      graph.set_op_version(version.value());
+    }
+  }
+
```

Comment:
Maybe we can save this logic (and the additional `version` argument to emitBuiltInCall) if we save version number in the graph in the first place and read it from there in all kinds of emission logic.